omniledger/service/service.go

```diff
@@ -891,9 +1084,11 @@ func (s *Service) save() {
 // deployments.
 func newService(c *onet.Context) (onet.Service, error) {
 	s := &Service{
-		ServiceProcessor: onet.NewServiceProcessor(c),
-		contracts:        make(map[string]OmniLedgerContract),
-		txBuffer:         newTxBuffer(),
+		ServiceProcessor:  onet.NewServiceProcessor(c),
+		contracts:         make(map[string]OmniLedgerContract),
+		txBuffer:          newTxBuffer(),
+		heartbeatsTimeout: make(chan string, 1),
```

Comment:
I expected to find heartbeats: newHeartbeats() here, because there are lots of calls to "if s.heatbeats.enabled()" that seem like they will fail if s.heartbeats is left nil. It is hard to see how this works when EnableViewChange() has not been called.

Reply:
`s.heartbeats` is a value not a reference. so it cannot be nil. the map inside it can be nil, which is what I use to check if it has been initialised.